### PR TITLE
Remove platform requirement from Package.swift

### PR DESCRIPTION
**Pull request by adam-fowler**
This PR removes the platform requirement from the Package.swift and replaces it with `@available` checks throughout the library where they weren't already placed.
I placed everything in the Benchmarks package inside a `DSLBenchmarks` enum so I only needed to add one `@available`.

These changes are required if Hummingbird is going to merge in its tracing middleware PR without making it a breaking change

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                "swift"
+            ],
+            "name": "Release _TracingBenchmarks",
+            "program": "${workspaceFolder:swift-distributed-tracing}/.build/release/_TracingBenchmarks",
+            "args": [],
+            "cwd": "${workspaceFolder:swift-distributed-tracing}",
+            "preLaunchTask": "swift: Build Release _TracingBenchmarks"
+        }
+    ]
+}
```
Comment (stevapple) [inline .vscode/launch.json:28]: Better excluded since it's going to be generated? Maybe it's worth adding `.vscode/` into `.gitignore` list.

**Review by stevapple — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (adam-fowler) [reply .vscode/launch.json:28]: Ah that was a mistake

Comment (adam-fowler) [reply .vscode/launch.json]: removed

**Review by slashmo — APPROVED** [review]

**Review by ktoso — APPROVED** [review]
Sorry for the delay here was out and sick for a while, thanks! LGTM

And great to hear Hummingbird adopting!

**Comment by ktoso** [comment]
@swift-server-bot add to allowlist

**Pull request merged**